manifests/install.pp

```diff
@@ -1,61 +1,199 @@
 class ss_solr::install inherits ss_solr {
-
-	Exec {
-		path => "/bin:/usr/bin:/usr/sbin",
+	if ss_solr::multithreaded{
+			$threads = $facts['processors']['count']
+	} else {
+			$threads = 1
 	}
+	if $facts['lsbdistcodename'] == 'jessie' {
+		notice("Your system is not compatible with this Puppet6 implementation. Please use the Puppet3 version.")
+	} else {
+		file { ['/usr/java', '/usr/share', '/usr/share/tomcat8', '/etc/tomcat8/']:
+			ensure => 'directory',
+			mode => '0755',
+			owner => 'root',
+			group => 'root',
+		}
+		file { '/usr/share/tomcat8/lib':
+			ensure => 'directory',
+			mode => '0755',
+			owner => 'tomcat8',
+			group => 'tomcat8',
+		}
+		# Install default jre to enable us to use update-alternatives correctly
+		package{"libecj-java":
+			ensure => installed
+		}
+		-> package{"default-jre":
+			ensure => installed
+		}
 
-	class { 'oracle_java':
-		version => $ss_solr::java_version,
-		add_alternative => true,
-		proxy_server => $ss_solr::http_proxy,
-		proxy_type => 'http',
-	}->
-	package { ['tomcat8', 'tomcat8-admin']:
-		ensure => 'present',
-	}->
-	archive { "/tmp/solr-$ss_solr::solr_version.tgz":
+		# Install latest java 8 by unpacking an archive from S3.
+		$install_path = '/usr/java'
+		$longversion = "jre1.8.0_261"
+		$version = "8u261"
+		$priority = 1000000 + 8 * 100000 + 261
+		archive { "/tmp/jre-8u261-linux-x64.tar.gz":
```

Comment:
Rather than introducing a new puppet/archive dependency can we switch these to using the existing servicetools module used on CWP and SC? https://github.com/silverstripeltd/puppet-servicetools/blob/master/manifests/install_deb.pp

Reply:
The dependency is nothing new and only replicates what the old `aco-oracle_java` module did to keep as much backwards compatibility as possible. So far that module included the `puppet/archive` module and used it internally.
I'll have a look if the `servicetools` would be an option to use.

Reply:
Looks like something we should use at some point. However `servicetools::install_deb` needs some work, because it fails to install the debian packages due to a failing `unless` check in exec, which seems to be a default in puppet that can't be changed in the install_deb so far.
From my point of view changing the implementation now would be risky for the planned release date unless we want to push that out.

Reply:
Sweet, I'm happy to leave this for now.